collaboration/extending-kyma-cli.md

```diff
@@ -0,0 +1,346 @@
+
+> We can use telepresence for local debugging. We can either use --swap-deployment or --new-deployment option.
+
+Telepresence would create a proxy in the kyma cluster and redirect the calls to local codebase. One can attach the debugger to his code base along with breakpoints to debug the code.
+
+##### Deploying and running the project
+
+The function generated by `init` command should be deployable onto kyma cluster. For deploying the file onto kyma cluster it would read the values from `config.yaml`. If the api name is not set then we should show some warning message and deploy.
+
+For running/debugging it locally we should provide makefile. The running/debugging commands should be part of readme.
+
+#### Working with deployed kyma cluster
+
+We should allow deployment of the functions on a k8s cluster. Following commands should be supported
+
+##### CRUD
+
+##### Commands that use config
+
+`kyma function apply -p <path-to-function-dir>`
+> This would create/update a function after reading the `config.yaml` from the path provided and to read the config like events, api etc. For deletion we should use the delete method.
+
+##### Expanded command with switches
+
+`kyma function create <function-name> -n <namespace> --runtime <runtime> --min-mem <mb> --max-mem <mb> --min-cpu <cpu> --max-cpu <cpu> --min-replicas 1 --max-replicas 1 --src-code <path-to-src-code> --dependecies <path-to-dep-file>`
+
+`kyma function update <function-name> -n <namespace> --runtime <runtime> --min-mem <mb> --max-mem <mb> --min-cpu <cpu> --max-cpu <cpu> --min-replicas 1 --max-replicas 1 --src-code <path-to-src-code> --dependecies <path-to-dep-file>`
+
+`kyma function delete <function-name> -n <namespace>`
+
+`kyma function get <function-name> -n <namespace>`
+
+*Labels*
+
+`kyma function label <function-name> ["foo=bar",..] -n <namespace>`
+
+*Expose*
+
+`kyma function expose <function-name> --secure(optional) --actions <GET/POST> -n <namespace>`
+
+*Service Catalog*
+
+`kyma function bind <function-name> --binding-instaces [<binding-instance-name>,...] -n <namespace>`
+`kyma function unbind <function-name> --binding-instaces [<binding-instance-name>,...] -n <namespace>`
+
+`kyma function bind <function-name> --binding-usage [<binding-usage-name>,...] -n <namespace>`
+`kyma function unbind <function-name> --binding-usage [<binding-usage-name>,...] -n <namespace>`
+
+*Bind to events*
+
+`kyma function trigger <function-name> --events [<event-name>,..] -n <namespace>`
+`kyma function trigger <function-name> --events [<event-name>,..] -n <namespace>`
+
+*Bind to environment variables*
+
+`kyma function bind <function-name> --env [<foo=bar>,..] -n <namespace>`
+`kyma function unbind <function-name> --env [<foo=bar>,..] -n <namespace>`
+
+*List all functions*
+
+`kyma function list -n <namespace>`
+
+> A detailed table view with function name, runtime, age, replicas and state
+
+*Runtimes*
+
+`kyma function get runtimes`
```

Comment:
In `kyma function init <function-name> --language <language-name>` language is used instead of runtime. Is this on purpose?

Reply:
fixed.